File: e2e-testing/page_objects/OpnameFormPage.js
const BasePage = require("./BasePage");

class OpnameFormPage extends BasePage {
  constructor(page) {
    super(page);
    this.modalTitleSelector = 'h3:has-text("Sesi Opname")';
    this.submitButton = 'button:has-text("Tutup Sesi & Apply")';
  }

  // Cek apakah modal Form Opname sudah terlihat
  async isFormVisible() {
    return await this.isElementVisible(this.modalTitleSelector);
  }

  // Cek apakah suatu item (misalnya "Kampas Rem") ada di dalam form modal
  async isItemDisplayed(itemName) {
    const itemLocator = this.page.locator(`tr:has-text("${itemName}")`);
    return await itemLocator.isVisible();
  }

  // Mengisi jumlah fisik aktual untuk item tertentu
  async enterPhysicalQuantity(itemName, quantity) {
    const row = this.page.locator("tr").filter({ has: this.page.locator("p", { hasText: itemName }) });
    const inputField = row.locator('input[type="number"]');
    await inputField.waitFor({ state: "visible" });
    await inputField.fill(quantity.toString());
  }

  // Mengisi catatan untuk item tertentu
  async enterNotes(itemName, notesText) {
    const row = this.page.locator("tr").filter({ has: this.page.locator("p", { hasText: itemName }) });
    const notesField = row.locator('input[type="text"]');
    await notesField.waitFor({ state: "visible" });
    await notesField.fill(notesText);
  }

  // Klik tombol simpan opname (Tutup Sesi & Apply)
  async clickSubmitOpname() {
    await this.click(this.submitButton);
  }
}

module.exports = OpnameFormPage;

File: e2e-testing/src/test/java/com/autoservice/pages/OpnameFormPage.java
package com.autoservice.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

/**
 * @file OpnameFormPage.java
 * @description Page Object Model for Active Stock Opname modal form in Java
 * @author Hafidz (QA Engineer 4)
 */
public class OpnameFormPage extends BasePage {
    private final By modalTitleSelector = By.xpath("//h3[contains(., 'Sesi Opname')]");
    private final By submitButton = By.xpath("//button[contains(., 'Tutup Sesi & Apply')]");

    public OpnameFormPage(WebDriver driver) {
        super(driver);
    }

    public boolean isFormVisible() {
        return isElementVisible(modalTitleSelector);
    }

    public boolean isItemDisplayed(String itemName) {
        By rowLocator = By.xpath("//tr[contains(., '" + itemName + "')]");
        return isElementVisible(rowLocator);
    }

    public void enterPhysicalQuantity(String itemName, int quantity) {
        By inputLocator = By.xpath("//tr[contains(., '" + itemName + "')]//input[@type='number']");
        fill(inputLocator, String.valueOf(quantity));
    }

    public void enterNotes(String itemName, String notesText) {
        By notesLocator = By.xpath("//tr[contains(., '" + itemName + "')]//input[@type='text']");
        fill(notesLocator, notesText);
    }

    public void clickSubmitOpname() {
        click(submitButton);
    }
}
